src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -537,9 +603,548 @@ namespace
+                                attrs.poolAttrs3d.includePaddingInAvgComputation)),
+                            geps[0],
+                            builder);
+                for (auto k = 0; k < 3; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[1],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs3d.windowShape[k]),
+                                gepStructOp,
+                                builder);
+                }
+                for (auto k = 0; k < 3; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[2],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs3d.windowStrides[k]),
+                                gepStructOp,
+                                builder);
+                }
+                for (auto k = 0; k < 3; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[3],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs3d.padBelow[k]),
+                                gepStructOp,
+                                builder);
+                }
+                for (auto k = 0; k < 3; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[4],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs3d.padAbove[k]),
+                                gepStructOp,
+                                builder);
+                }
+                break;
+            }
+            case AttrsType::CONV1D:
+            {
+                auto conv1dTy = getLLVMType(AttrsType::CONV1D, llvmDialect);
```

Comment:
Let's try to improve this a bit by refactoring all the switch cases to utility functions and leave this function as a dispatcher. Something like:
```
// Doc showing what is initializing exactly.
static initConv2D(...){
   ...
}

switch(...) {
  case AttrsTrype::CONV1D:
      initConv1D(...);
  case YYY:
      initYYY
}
```